### DISCUSSION TITLE
Is there a better way for one object has multiple many-to-many relationship definition?

### DISCUSSION THREAD

**Discussion by wangxin688** [sqlalchemy/sqlalchemy#9513] [Usage Questions] [answered]
A real world case:
First, I defined a object named `Contact`, it has multiple many-to-many relationship with other objects, should I define a associated table for every object? Or there's a more elegant way for this case?
for code bellow, I need to define a new table again and again after add object related to `Contact`. I'm not sure if this is the right way. Thanks a lot
```python3
class SiteContact(Base):
    __tablename__ = "dcim_site_contact_link"
    __multi_tenant__ = False
    site_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("dcim_site.id"), primary_key=True, comment="site fk_id")
    contact_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("contact.id"), primary_key=True, comment="contact fk_id")


class CircuitContact(Base):
    __tablename__ = "circuit_contact_link"
    __multi_tenant__ = False
    circuit_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("circuit.id"), primary_key=True, comment="circuit fk_id")
    contact_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("contact.id"), primary_key=True, comment="contact fk_id")

class Contact(Base, ):
    __tablename__ = "contact"
    id: Mapped[uuid_pk]
    name: Mapped[str_50]
    email: Mapped[str_50_optional]
    phone: Mapped[str_50_optional]
    obj: Mapped[str_50]
    dcim_site: Mapped[List["Site"]] = relationship(
        "Site", secondary="dcim_site_contact_link", overlaps="contact", back_populates="contact"
    )
    circuit: Mapped[List["Circuit"]] = relationship(
        "Circuit", secondary="circuit_contact_link", overlaps="contact", back_populates="contact"
    )
````

**Comment by CaselIT** [comment, marked as answer]
Hi,

If you want to use foreign keys at the database level I don't think there is another solution for this.
Are you ok with solutions that would remove the foreign keys on the database?

**Reply by wangxin688** [reply]
Could you give a example without forienkey? It's OK if reach the same goal.

**Reply by lelit** [reply]
Here it is:
https://docs.sqlalchemy.org/en/20/_modules/examples/generic_associations/generic_fk.html

I used something very similar in the past but, while it is quite handy, in the end I prefer to leverage real FKs, paying the "price" of maintaining all those little many-to-many secondary tables.

bye, lele.

**Reply by CaselIT** [reply]
did not know there was an example. what I had in mind what something like this though!
